### PR TITLE
Updating with Patch method on articles endpoint.

### PR DESCRIPTION
**Pull request by bayardt**

### CODE REVIEW

File: __tests__/endpoints.test.js [hunk 1]
```diff
@@ -58,4 +58,41 @@ describe("/api/articles/:article_id", () => {
...
+        .send({ change_votes: 15 })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Bad Request");
+        });
+    });
+    test("should return 400 when submitting invalid value.", () => {
+      return request(app)
+        .patch("/api/articles/10")
+        .send({ inc_votes: 'banana' })
```
Comment (popatre) [inline __tests__/endpoints.test.js]: Test - what of there are extra keys i.e. title, as well. 

Presume the test will handle it, but good to have a test for it to show its been considered

File: __tests__/endpoints.test.js [hunk 2]
```diff
@@ -58,4 +58,41 @@ describe("/api/articles/:article_id", () => {
...
+            body: expect.any(String),
+            topic: "mitch",
+            created_at: expect.any(String),
+            votes: 15,
+          });
+        });
+    });
+    test("should return 400 when submitting invalid key.", () => {
+      return request(app)
+        .patch("/api/articles/10")
```
Comment (popatre) [inline __tests__/endpoints.test.js:81]: Test - Patch to an article that doesnt exist?
test - patch to invalid article id

File: models/articles.models.js [hunk 3]
```diff
@@ -16,3 +16,20 @@ exports.selectArticleById = (requestedArticleId) => {
...
+  return db
+    .query(
+      `UPDATE articles SET votes = votes + ${voteCount} WHERE article_id = ${requestedArticleId} RETURNING *;`,
+    )
+    .then(({ rows }) => {
+      const article = rows[0];
+      if (!article) {
+        return Promise.reject({
+          status: 404,
+          msg: `No article found for article_id: ${requestedArticleId}`,
```
Comment (popatre) [inline models/articles.models.js:31]: Is there a test for this?
I'm only seeing whats in this pull request, so can only see the test for the patch stuff.

**Review by popatre — CHANGES_REQUESTED** [review]

**Review by Angaca — APPROVED** [review]
Good to go!

**Pull request merged**